hpx/runtime/applier/apply_helper.hpp

```diff
@@ -116,39 +115,78 @@ namespace hpx { namespace applier { namespace detail
         }
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    HPX_FORCEINLINE bool needs_to_execute_asynchronously()
```

Comment:
Would it make sense to put this function into it's own header? We use the same functionality at three different places now. This would reduce code duplication and sources for errors.